tests/Providers/FindologicServiceProviderTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Findologic\Tests\Providers;
+
+use Findologic\Services\SearchService;
+use Plenty\Plugin\ConfigRepository;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+use Findologic\Providers\FindologicServiceProvider;
+use Findologic\Constants\Plugin;
+use Plenty\Plugin\Events\Dispatcher;
+use Plenty\Plugin\Http\Request;
+
+/**
+ * Class SearchServiceTest
+ * @package Findologic\Tests
+ */
+class FindologicServiceProviderTest extends TestCase
+{
+    /**
+     * @var SearchService|MockObject
+     */
+    protected $searchService;
+
+    /**
+     * @var Request|MockObject
+     */
+    protected $request;
+
+    /**
+     * @var Dispatcher|MockObject
+     */
+    protected $eventDispatcher;
+
+    /**
+     * @var ConfigRepository|MockObject
+     */
+    protected $configRepository;
+
+    public function setUp()
+    {
+        $this->searchService = $this->getMockBuilder(SearchService::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->request = $this->getMockBuilder(Request::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->eventDispatcher = $this->getMockBuilder(Dispatcher::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->configRepository = $this->getMockBuilder(ConfigRepository::class)->disableOriginalConstructor()->setMethods([])->getMock();
+    }
+
+    public function testBootShopKeyNotSet()
+    {
+        $this->configRepository->expects($this->once())->method('get')->willReturn('');
+
+        $this->eventDispatcher->expects($this->never())->method('listen');
+
+        $this->runBoot();
+    }
+
+    public function testBootShopKeySetAndAliveTestFails()
+    {
+        $this->configRepository->expects($this->once())->method('get')->willReturn('testShopKey');
+
+        $this->eventDispatcher->expects($this->never())->method('listen');
+
+        $this->runBoot();
+    }
+
+    public function testIsNotSearchPageAndIsNotActiveOnCatPage()
+    {
+        $this->configRepository->expects($this->any())->method('get')->willReturnMap([
+            [
+                Plugin::CONFIG_SHOPKEY, false, 'testShopKey'
+            ]
+        ]);
+
+        $this->searchService->expects($this->once())->method('isAliveTest')->willReturn(true);
+
+        $this->eventDispatcher->expects($this->at(0))->method('listen')->with('IO.Resources.Import');
```

Comment:
Try to use `withConsecutive` instead of providing a call index, as this can give us a hard time in the future.